javascript/javascript2/week3/index.js

```diff
@@ -0,0 +1,198 @@
+//1.Log out the text Called after 2.5 seconds
+window.addEventListener("load", loaded);
+function loaded() {
+    setTimeout(logout, 2500)
+};
+function logout() {
+    console.log("Called after 2.5 second");
+};
+//2.Create a function that takes 2 parameters
+function logoutString(delay, stringToLog) {
+    setTimeout(() => {
+        console.log(stringToLog);
+    }, delay * 1000)
+}
+logoutString(5, "This string logged after 5 seconds");
+logoutString(3, "This string logged after 3 seconds");
+
+
+//3.Create a button in html after5seconds
+const clickButton = document.getElementById("myButton")
+clickButton.addEventListener("click", after5seconds)
+function after5seconds() {
+    logoutString(5, "Called after 5 seconds")
+}
+//4 Create two functions and call them
+const earthLogger = () => console.log("Earth");
+const SaturnLogger = () => console.log("Saturn");
+function PlanetLogFunction(planetLogFunction) {
+    planetLogFunction();
+}
+PlanetLogFunction(earthLogger);
+PlanetLogFunction(SaturnLogger);
+//5 location
+let button = document.getElementById("get-locationa");
+let latText = document.getElementById("latitude");
+let longText = document.getElementById("longitude");
+
+button.addEventListener("click", function () {
+    navigator.geolocation.getCurrentPosition(function (position) {
+        let lat = position.coords.latitude;
+        let long = position.coords.longitude;
+
+        latText.innerText = lat.toFixed(2);
+        longText.innerText = long.toFixed(2);
+    });
+});
+
+//6 location in map
+let buttonQ5 = document.getElementById("get-location");
+let locationlink = document.querySelector('#locationlink');
+buttonQ5.addEventListener("click", function () {
+    navigator.geolocation.getCurrentPosition(function (position) {
+        let lat = position.coords.latitude;
+        let long = position.coords.longitude;
+        locationlink.href = `https://maps.google.com/maps?q=${lat},${long}&hl=es;z=14&amp;output=embed`;
+        locationlink.innerText = `click here to see where are you in the map`
+    });
+});
+
+
+7
+function runAfterDelay(delay, callback) {
+    setTimeout(callback, delay * 1000);
+}
+runAfterDelay(4, function () {
+    console.log('should be logged after 4 second');
+});
+runAfterDelay(2, function () {
+    console.log(9 + 3);
+});
+
+//8 
+let clickCountArray = [];
+const doubleClick = document.getElementById("doubleClick");
+doubleClick.addEventListener("click", doubleClickFunction);
+function doubleClickFunction() {
+    clickCountArray.push("click");
+    if (clickCountArray.length > 1) {
+        console.log("double click!");
+    } else if (clickCountArray.length = 1) {
+        console.log('clicked once');
+    };
+    //0.5s later execute this function
+    setTimeout(() => {
+        doubleClick.removeEventListener("click", doubleClickFunction)
+    }, 500);
+};
```

Comment:
Something more easier..there is a doubleclick function for eventlistener https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event

Reply:
thanks